code/modules/mod/modules/modules_engineering.dm

```diff
@@ -151,3 +151,100 @@
 	QDEL_NULL(chain)
 	return ..()
 
+///Atmos water tank module
+
+/obj/item/mod/module/firefighting_tank
+	name = "MOD firefighting tank"
+	desc = "A refridgerated and pressurized module tank with an extinguisher nozzle, intended to fight fires. Swaps between extinguisher, nanofrost launcher, and metal foam dispenser for breaches. Nanofrost converts plasma in the air to nitrogen, but only if it is combusting at the time."
+	icon_state = "firefighting_tank"
+	module_type = MODULE_ACTIVE
+	complexity = 2
+	active_power_cost = DEFAULT_CHARGE_DRAIN * 3
+	device = /obj/item/extinguisher/mini/mod
+
+#define EXTINGUISHER 0
+#define NANOFROST 1
+#define METAL_FOAM 2
+
+/obj/item/extinguisher/mini/mod
+	name = "modsuit extinguisher nozzle"
+	desc = "A heavy duty nozzle attached to a modsuit's internal tank."
+	icon = 'icons/obj/watertank.dmi'
+	icon_state = "atmos_nozzle_1"
+	item_state = "nozzleatmos"
+	safety = 0
+	max_water = 500
+	power = 8
+	precision = 1
+	cooling_power = 5
+	w_class = WEIGHT_CLASS_HUGE
+	flags = NODROP //Necessary to ensure that the nozzle and tank never seperate
+	var/nozzle_mode = EXTINGUISHER
+	var/metal_synthesis_charge = 0
```

Comment:
Hmm, I think it may be better to have this set at the maximum, then start decreasing
```suggestion
	var/metal_synthesis_charge = 5
```

Reply:
Starting at 5 breaks the current implementation as the timer doesn't start until foam is used.